R/show_MariaDBConnection.R

```diff
@@ -3,16 +3,23 @@
 #' @usage NULL
 show_MariaDBConnection <- function(object) {
   info <- dbGetInfo(object)
-  cat("<MariaDBConnection>\n")
+  cat("<", class(object), ">\n", sep = "")
   if (dbIsValid(object)) {
-    cat("  Host:    ", info$host, "\n", sep = "")
-    cat("  Server:  ", info$serverVersion, "\n", sep = "")
-    cat("  Client:  ", info$client, "\n", sep = "")
-    # cat("  Proto:   ", info$protocolVersion, "\n", sep = "")
-    # cat("  ThreadId:", info$threadId, "\n", sep = "")
-    # cat("  User:    ", info$user, "\n", sep = "")
-    # cat("  ConType: ", info$conType, "\n", sep = "")
-    # cat("  Db:      ", info$dbname, "\n", sep = "")
+    cat("  Hostname:       ", info$host, "\n", sep = "")
+    cat("  Username:       ", info$user, "\n", sep = "")
+    cat("  Database:       ", info$dbname, "\n", sep = "")
+    cat("  Server Port:    ", info$port, "\n", sep = "")
+    cat("  Connection:     ", info$con.type, "\n", sep = "")
+    cat("  Protocol:       ", info$protocol.version, "\n", sep = "")
+    if (!is.null(info$ssl.cipher)) {
+      cat("  SSL Cipher:     ", info$ssl.cipher, "\n", sep = "")
+    }
+    cat("  Thread Id:      ", info$thread.id, "\n", sep = "")
+    cat("  Client Version: ", info$client.version, "\n", sep = "")
+    cat("  Client Flags:   ", format(as.hexmode(info$client.flag), width=8, upper.case=T), "\n", sep = "")
+    cat("  Server Version: ", info$db.version, "\n", sep = "")
+    cat("  Capabilities:   ", format(as.hexmode(info$server.capabilities), width=8, upper.case=T), "\n", sep = "")
+    cat("  SQL Status:     ", format(as.hexmode(info$status), width=8, upper.case=T), "\n", sep = "")
```

Comment:
This seems too much when printing the connection. It might be enough to keep this info exclusive to `dbGetInfo()` .

Reply:
Sure, I think it would be fine to cut this back if you want.   I was just under the impression that most of the time, people show an object (and specifically this connection object) when it's for troubleshooting purposes.  In which case, most of this information is very handy.   For example, the Thread ID (the server references it as the Session ID) is very useful on the server-side as it identifies the session when using `SHOW PROCESSLIST`.   So, I would argue that one specifically should be kept.  However, the versions and flags could certainly be suppressed.

I would be fine with something more terse like the following:
```
<MySQLConnection>
  Connection: r_user@my-database-1.xyzzy.us-east-1.rds.amazonaws.com<r_test_db> via TCP/IP over SSL
  Session Id: 10615
```
Let me know if that's what you'd like and I'll update the PR.

Reply:
Yeah, after re-working it to the single line, I liked it better too!